pkg/ebpf/c/tracee.bpf.c

```diff
@@ -6186,14 +6185,21 @@ int sched_process_exec_signal(struct bpf_raw_tracepoint_args *ctx)
     struct task_struct *task = (struct task_struct *) ctx->args[0];
     if (task == NULL)
         return -1;
+    struct task_struct *leader = get_leader_task(task);
+    struct task_struct *parent = get_leader_task(get_parent_task(leader));
+
+    // The hash is always calculated with "task_struct->pid + start_time".
+    u32 task_hash = hash_task_id(get_task_host_tgid(task), get_task_start_time(task));
```

Comment:
got rid of bpf_get_current_pid_tgid() call (no need).

Reply:
I don't understand.
This should be the current task unique ID (hash).
But you get the PID, which will be the same as the leader - `get_task_host_tgid(task)` == `get_task_host_tgid(leader)`.

Reply:
I don't understand.
This should be the current task unique ID (hash).
But you get the PID, which will be the same as the leader - `get_task_host_tgid(task)` == `get_task_host_tgid(leader)`.

Reply:
Oh god, yep, you are totally right Alon. Funny because when doing it I did "I need the PID" (no clue why I thought it). So I did it wrong consciently, shame on me. Fixing it.

Reply:
```C
    // The hash is always calculated with "task_struct->pid + start_time".
    u32 task_hash = hash_task_id(get_task_host_pid(task), get_task_start_time(task));
    u32 parent_hash = hash_task_id(get_task_host_pid(parent), get_task_start_time(parent));
    u32 leader_hash = hash_task_id(get_task_host_pid(leader), get_task_start_time(leader));
```

in both: sched_process_exec and sched_process_exit.